modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase

```diff
@@ -301,6 +433,98 @@ definition {
 			actionName = "New Action Update");
 	}
 
+	@description = "LPS-146871 - Verify that it's possible to edit an Action with Groovy Script"
+	@priority = "5"
+	test CanEditActionWithGroovyScript {
+		property portal.acceptance = "true";
+
+		task ("Given a Groovy Script") {
+			var groovyScript = '''
+			import com.liferay.object.service.ObjectEntryLocalServiceUtil;
+			import com.liferay.object.model.ObjectEntry;
+			import com.liferay.portal.kernel.service.ServiceContext;
+
+			import java.io.Serializable;
+			import java.util.Map;
+
+			ObjectEntry objectEntry = ObjectEntryLocalServiceUtil.getObjectEntry(objectEntryId);
+
+			Map<String, Serializable> values = objectEntry.getValues();
+
+			values.put("customObjectFieldActionTest", "Action Test Works")
+
+			System.out.println(objectEntry);
+
+			ObjectEntryLocalServiceUtil.updateObjectEntry(objectEntry.getUserId(), objectEntryId, values, new ServiceContext());
+			''';
+		}
+
+		task ("Given an object, two fields and an entry is created") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object",
+				objectName = "CustomObject",
+				pluralLabelName = "Custom Objects");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "true",
+				objectName = "CustomObject");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field Action Test",
+				fieldName = "customObjectFieldActionTest",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject");
+
+			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+				fieldName = "customObjectField",
+				objectName = "CustomObject",
+				value = "Entry Test");
+		}
+
+		task ("When an action is created using script groovy") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.addObjectActionViaUI(
+				actionName = "Custom Action",
+				active = "true",
+				objectLabel = "Custom Object",
+				scriptValidation = "${groovyScript}",
+				thenAction = "Groovy Script",
+				whenAction = "On After Add");
+		}
```

Comment:
We can combine this into one task: `Given a Grooy Script, an object with two text fields, an object entry, and an Action using Groovy Script are created`